doc/architecture/decisions/0004-reuse-api-for-cas1-and-cas3.md

```diff
@@ -0,0 +1,86 @@
+
+Date: 2022-10-22
+
+## Status
+
+Accepted
+
+## Context
+
+CAS1 (Approved Premises) and CAS3 (Temporary Accommodation) share a common
+business domain - they both manage accommodation in various properties, match
+potential residents, and manage the arrival and departure of residents. They
+also both accept referrals to their services from Probation Practitioners.
+
+Both services break down their workflow into a common pattern: Apply, Assess,
+Match, Manage, and Report.
+
+CAS1 have been working on this API for the last few months and have developed a
+minimal viable property management service called mini-manage. The API includes
+development tooling and has infrastructure in place to run it in a live
+environment. CAS3 have a short 3 month deadline to go live with their own
+property management service.
+
+In order to increase the odds of success for CAS3 and to save time and money,
+CAS3 have been exploring [1,2] what software can be reused in collaboration with
+CAS1.
+
+Both CAS teams are currently procured separately so we cannot merge to form one
+API team/squad. There are longer term ambitions that this will eventually be the
+case.
+
+1. One option was to fork the API now and develop a separate set of services.
+This would provide CAS3 a lot of reuse in the short term but would close the
+door to the long term benefits of having a single service. We believed that it
+would be highly unlikely to ever be cost effective to merge back those two
+services back together in the future.
+2. The second and preferred option was to attempt to work within the same API,
+accept the impact to delivery in the short term in an attempt gain the longer
+term benefits for MoJ. Taking this path would not block the path to forking in
+future should the trade-offs change.
+
+Our proposal for option 2[1] was taken to our joint service owner for approval
+on October 13th 2022.
+
+[1]
+<https://docs.google.com/document/d/1zX28jObx2MCt57mafi-d4GE4vO-UAexLs_Ly2KQfU3U/edit?pli=1>
+
+[2]
+<https://docs.google.com/document/d/1vclqsYO0BRlMWTfp5rA0Ev-GnK82VsU9d7Si07mWlqo/edit#heading=h.4hk4dnxkopf9>
+
+## Decision
+
+Both CAS teams will aim to share and build upon the same API.
+
+Relatedly there is a frontend repository[3] that CAS3 will fork and develop
+independently.
+
+[3] <https://github.com/ministryofjustice/hmpps-approved-premises-ui>
+
+## Consequences
+
+- a single technical architect that's shared across both projects to maintain
+  oversight and improve the chances of delivering a joined up service
+- both teams may have to collaborate and design parts of the service together,
+  hopefully the tech leads and technical architect can do this for the most part
+  to protect the rest of the teams time
+- we may find that working on the same API proves too logistically difficult in
+  which case CAS3 could fork away at that point
+- we may find there isn't as much shared behaviour as we anticipate, in which
+  case CAS3 could work away at that point
```

Comment:
should this be `fork away` rather than `work away`?

Reply:
Yup, thanks for the spot!